stock_location_children/models/stock_location.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import operator
+
+from odoo import api, fields, models
+from odoo.fields import Command
+
+
+class StockLocation(models.Model):
+
+    _inherit = "stock.location"
+
+    children_ids = fields.Many2many(
+        comodel_name="stock.location",
+        relation="stock_location_children_ids",
+        column1="parent_id",
+        column2="children_id",
+        compute="_compute_children_ids",
+        store=True,
+        # Don't put recursive here as the query in compute method is sufficent
+        help="All the children (multi-level) stock location of this location",
+    )
+
+    @api.depends("child_ids", "children_ids.child_ids")
+    def _compute_children_ids(self):
+
+        query = """SELECT sub.id, ARRAY_AGG(sl2.id) AS children
+            FROM stock_location sl2,
+            (
+            SELECT id, parent_path
+            FROM stock_location sl
+            ) sub
+            WHERE sl2.parent_path LIKE sub.parent_path || '%%'
+            AND sl2.id != sub.id
+            AND sub.id IN %s
+            GROUP BY sub.id;
+        """
```

Comment:
Maybe more readable than sub and sl2:
```suggestion
        query = """
            SELECT sl.id, ARRAY_AGG(child.id) AS children
            FROM stock_location sl
            JOIN stock_location child
            ON child.parent_path LIKE sl.parent_path || '_%%'
            AND sl.id IN %s
            GROUP BY sl.id
        """
```

Reply:
Yes, I didn't want to modify things that worked before.